python/ray/util/sgd/v2/backends/backend.py

```diff
@@ -1,45 +1,182 @@
-from typing import Callable
+import logging
+from typing import Callable, Iterator, TypeVar, List, Any
 
+import ray
+from ray.exceptions import RayActorError
 from ray.util.sgd.v2.worker_group import WorkerGroup
 
+T = TypeVar("T")
+
+logger = logging.getLogger(__name__)
+
+
+def fetch_next():
+    """Fetch next item from result queue on each worker."""
+    # TODO: Implement along with sgd.report()
+    return None
+
+
+def fetch_all():
+    """Fetch all items from result queue on each worker."""
+    # TODO: Implement along with sgd.report()
+    return [None]
+
 
 class BackendConfig:
+    """Parent class for configurations of backends (torch, horovod, etc.)"""
+
     @property
     def backend_name(self):
         raise NotImplementedError
 
     def validate(self, name):
-        assert name == self.backend_name
+        assert name == self.backend_name(), (name, self.backend_name)
 
 
 class BackendExecutor:
+    """Main execution class for SGD backends (torch, tensorflow, etc.).
```

Comment:
Does this make sense to call them SGD backends or something more generic like training backends?

Reply:
yeah training backends sounds good.